src/dotnet/Program.cs

```diff
@@ -45,7 +44,7 @@ public static int Main(string[] args)
 
         }
 
-        private static int ProcessArgs(string[] args)
+        public int ProcessArgs(string[] args, ITelemetry telemetryClient)
```

Comment:
Looks like it is public so it can be unit tested. How about using InternalsVisibleTo instead?